// jest-compact-reporter/dist/reporter/JestCompactReporter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JestCompactReporter = void 0;
const reporters_1 = require("@jest/reporters");
const utils_1 = require("../utils");
class JestCompactReporter extends reporters_1.BaseReporter {
    constructor(globalConfig, options) {
        super();
        this.globalConfig = globalConfig;
        this.options = {
            diffs: options.diffs === undefined ? true : options.diffs,
            colours: options.colours === undefined ? true : options.colours,
            showPassingTests: options.showPassingTests === undefined
                ? false
                : options.showPassingTests,
        };
        global.reporterGlobals = { coloursEnabled: this.options.colours };
    }
    onRunStart() {
        utils_1.printHeaderOnStart();
    }
    onRunComplete(contexts, results) {
        utils_1.printHeaderOnComplete();
        utils_1.printPassedSuites(results, this.options.showPassingTests);
        this.options.diffs && utils_1.printFailedTestDiffs(results);
        utils_1.printFailedSuites(results, this.options.showPassingTests);
        results.snapshot.failure &&
            utils_1.printUncheckedSnapshotsSummary(results.snapshot);
        utils_1.printSummary(results);
    }
}
exports.JestCompactReporter = JestCompactReporter;
